### PR TITLE
Upgrade to support electron 5

### PR DESCRIPTION
**Pull request by alexstrat**
Based on https://github.com/electron/spectron/pull/361

## Todos
- [x] update documentation: `nodeIntegration` +  versions
- [x] Figure out failing tests with `Cannot read property 'isLoading' of undefined`
- [x] Fix failing test _"gets the render process console logs and clears them"_
- [x] Figure out failing tests with `invalid session id`

### CODE REVIEW

**Comment by codebytere** [comment]
@alexstrat happy to take a look when this is in a reviewable state!

File: test/accessibility-test.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ var it = global.it
 var beforeEach = global.beforeEach
 var afterEach = global.afterEach
 
-describe('app.client.auditAccessibility()', function () {
```
Comment (alexstrat) [inline test/accessibility-test.js]: ⚠️To fix and remove me before merging

File: test/accessibility-test.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ var it = global.it
...
+xdescribe('app.client.auditAccessibility()', function () {
```
Comment (alexstrat) [inline test/accessibility-test.js]: ⚠️To fix and remove me before merging

**Comment by alexstrat** [comment]
@codebytere I think we're good here, ready for review! :)
Worth reviewing, merging and releasing  #361 before :)

**Comment by stevebaxter** [comment]
Is there any possibility of getting this merged and released soon? We've just updated to Electron 5 and our Spectron tests no longer work on Windows when running under Jenkins (our application never seems to start and the Jenkins job eventually times out).

**Comment by codebytere** [comment]
cc @MarshallOfSound for secondary 👀, but then i think we can get this merged and released.

**Comment by codebytere** [comment]
@alexstrat i've merged and released compatibility for `v4.x.y`: can you rebase this and we'll get it merged and released as well? 

Thanks again for doing the work on this 🙇‍♀ it's very much appreciated!

**Comment by alexstrat** [comment]
@codebytere rebased ✅

**Comment by michalzaq12** [comment]
:clap: clap

**Pull request merged**